app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/LiveDataFragment.java

```diff
@@ -0,0 +1,438 @@
+                                        new VideosInSameCategoryViewModel.Factory(
+                                                getActivity().getApplication(), category);
+
+                                // view model will not be re-created as long as the lifecycle owner
+                                // lifecycle observer and tag doesn't change
+                                VideosInSameCategoryViewModel viewModel =
+                                        ViewModelProviders.of(mFragmentActivity, factory).get(
+                                                category, VideosInSameCategoryViewModel.class);
+
+                                viewModel.getVideosInSameCategory().observe(mLifecycleOwner,
+                                        new Observer<List<VideoEntity>>() {
+                                            @Override
+                                            public void onChanged(
+                                                    @Nullable List<VideoEntity> videoEntities) {
+                                                if (videoEntities != null) {
+                                                    startEntranceTransition();
+                                                    adapter.setItems(videoEntities,
+                                                            new Comparator<VideoEntity>() {
+                                                                @Override
+                                                                public int compare(VideoEntity o1,
+                                                                                   VideoEntity o2) {
+                                                                    return o1.getId() == o2.getId() ? 0 : -1;
+                                                                }
+                                                            }, new Comparator<VideoEntity>() {
+                                                                @Override
+                                                                public int compare(VideoEntity o1,
+                                                                                   VideoEntity o2) {
+                                                                    return o1.equals(o2) ? 0 : -1;
+                                                                }
+                                                            });
+                                                }
+                                            }
+                                        });
+
+                                // create current category row
+                                ListRow row = new ListRow(new HeaderItem(categoryEntity.getCategoryName()),
+                                        adapter);
+                                rows.add(row);
+                            }
+
+                            mRowsAdapter.setItems(rows, new Comparator<ListRow>() {
```

Comment:
if you call setItems() with new a new set of ListRows, it seems like there is no place you un-observe the previous rows' observer registerd at line 296

Reply:
This is also one of the key problems I have question on LiveData, @yigit do you have comments about what is the best practice?